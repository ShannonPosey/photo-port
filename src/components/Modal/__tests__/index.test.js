import React from "react";
import {render, cleanup, fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Modal from "..";

const mockToggleModal = jest.fn();
const currentPhoto = {
    name: "Park bench",
    category: "landscape",
    description: "lorem ispsum dolor sit amet.",
    index: 1
};

afterEach(cleanup);

describe("Modal component", () => {
    it("renders", () => {
        // baseline
        render(<Modal onClose={mockToggleModal} currentPhoto={currentPhoto} />);
    });
    // snapshot test
    it("matches snapshot DOM nod structure", () => {
        const {asFragment} = render(<Modal onclose={mockToggleModal} currentPhoto={currentPhoto}/>)

        expect(asFragment()).toMatchSnapshot()
    });
});

describe('Click Event', () => {
  it('calls onClose handler', () => {
      // Arrange: Render Modal
    const { getByText } = render(<Modal onClose={mockToggleModal} currentPhoto={currentPhoto}/>);
    // Act: Simulate click event
    fireEvent.click(getByText('Close this modal'));
    // Assert: Expected matcher
    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
})  