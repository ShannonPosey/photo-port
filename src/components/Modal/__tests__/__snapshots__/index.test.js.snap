// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM nod structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park bench
      </h3>
      <img
        alt="current category"
      />
      <p>
        lorem ispsum dolor sit amet.
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
